Objects/cown.c

```diff
@@ -174,6 +185,13 @@ int _PyCown_is_released(PyObject *self) {
     return STATE(cown) == Cown_RELEASED;
 }
 
+int _PyCown_is_pending_release(PyObject *self) {
+    assert(Py_TYPE(self) == &PyCown_Type && "Is pending release called on non-cown!");
```

Comment:
I feel like an assert is not enough since asserts might be disabled:
```suggestion
    if (Py_TYPE(self) != &PyCown_Type) {
        assert(false && "`_PyCown_is_pending_release` called on non-cown!");
        PyErr_SetString(PyExc_RuntimeError, ""`_PyCown_is_pending_release` called on non-cown!");
        return false;
    }    
```

(Not a blocker for this PR, but a comment before we want to get this into cpython)

Reply:
Since this is not a user-facing function, I think we should leave it at the assert. If it is called with a non-cown, that's a C error that should not appear as an exception in Python-land.